### PR TITLE
Throw an error when css strings is used with styled components

### PR DESCRIPTION
**Pull request by terrysahaidak**
Fixes RNBW-2466

## What changed (plus any additional context for devs)

Since our own faster implementation of styled components doesn't support CSS strings - we should make sure no one is using it. So from now on, it will throw an error if template literal syntax is used.

So this won't work:
```
const Container = styled.View``;
```

And this will work
```
const Container = styled.View({});
```

When we call the function with template literals instead of parenthesis 
```
syled`` 
```

it passes the template string and all the string interpolations (`${}`) as an array. So I check for an array to detect if we used template literal.

## PoW (screenshots / screen recordings)

image
image

## Dev checklist for QA: what to test

It should not break anything

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

File: src/styled-thing/index.js [hunk 1]
```diff
@@ -97,6 +97,12 @@ export default function styled(Component) {
   }
 
   function StyledComponentFactory(styles) {
+    if (Array.isArray(styles)) {
```
Comment (osdnk) [inline src/styled-thing/index.js:100]: Can we make this only in DEV?

Comment (terrysahaidak) [reply src/styled-thing/index.js:100]: Only as "performance improvements" since it will throw some error anyway

Comment (osdnk) [reply src/styled-thing/index.js:100]: yes, that's what I mean. Less code to run on prod = better

Comment (terrysahaidak) [reply src/styled-thing/index.js]: Agreed

**Review by skylarbarrera — APPROVED** [review]

**Review by osdnk — APPROVED** [review]

**Pull request merged**